string node
---
string[] subscribing
string[] publishing
string[] services